File: back/src/main/java/com/example/back/entity/DailyExchange.java
package com.example.back.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "daily_exchange")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class DailyExchange {
    @Id
    @Column(name = "date")
    private LocalDate date;

    private Double dollar;
    private Double gold;
    private Double oil;
    private Double dxy;
    @Column(name = "base_rate")
    private Double baseRate;
    @Column(name = "us_base_rate")
    private Double usBaseRate;
}

File: front/src/api/dailyExchange.ts
// 일별 집계 환율/금/유가/지수/금리 API
import axios from 'axios';

export interface DailyExchange {
  date: string; // yyyy-MM-dd
  dollar: number;
  gold: number;
  oil: number;
  dxy: number;
  baseRate: number;
  usBaseRate: number;
}

// 가장 최근 1일치 데이터 조회
export async function fetchLatestDailyExchange(): Promise<DailyExchange | null> {
  const res = await axios.get('/api/daily-exchange/latest');
  return res.data ?? null;
}

// (추후 확장) 기간별 데이터 조회 등도 추가 가능
